pallets/msa/src/types.rs

```diff
@@ -30,6 +35,39 @@ pub struct AddKeyData<T: Config> {
 	pub new_public_key: T::AccountId,
 }
 
+impl<T: Config> EIP712Encode for AddKeyData<T> {
+	fn encode_eip_712(&self) -> Box<[u8]> {
+		lazy_static! {
+			// get prefix and domain separator
+			static ref PREFIX_DOMAIN_SEPARATOR: Box<[u8]> =
+				get_eip712_encoding_prefix("0xcccccccccccccccccccccccccccccccccccccccc");
```

Comment:
nit: seems like this could be refactored, as it's in more than one place

Reply:
So it depends. I kept them separated because in EIP-712 the idea is that each one of them (at least for each pallet) we will have different contract addresses. We can make a decision that maybe we would like to simplify everything and just use the same contract address or not but that is a bigger decision.